inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -94,7 +94,16 @@ bool InLaneCruisingPlugin::plan_trajectory_cb(cav_srvs::PlanTrajectoryRequest& r
   trajectory.trajectory_points = compose_trajectory_from_centerline(downsampled_points, req.vehicle_state); // Compute the trajectory
   trajectory.initial_longitudinal_velocity = std::max(req.vehicle_state.longitudinal_vel, config_.minimum_speed);
 
-  resp.trajectory_plan = trajectory;
+  if (config_.enable_object_avoidance)
+  {
+    object_avoidance::ObjectAvoidance obj;
```

Comment:
Is there a reason to create and recreate this on a timestep-by timestep basis? I think we could just initialize it or set it to `nullptr` in the constructor (or wherever this param is loaded) and then check here to see if it's non-null.

Reply:
Either that or make it a free function and avoid having to create an object at all, if there's no state here that matters.

Reply:
addressed, but did not add it to the constructor because that would fail all the ilc unit tests.